mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -100,12 +100,20 @@ class LogisticRegression(override val uid: String)
   def setThreshold(value: Double): this.type = set(threshold, value)
   setDefault(threshold -> 0.5)
 
-  override protected def train(dataset: DataFrame): LogisticRegressionModel = {
-    // Extract columns from data.  If dataset is persisted, do not persist oldDataset.
+  private var optInitialWeights: Option[Vector] = None
+  /** @group setParam */
+  def setInitialWeights(value: Vector): this.type = {
```

Comment:
Let's have it private for now. Since for multinomial one, the initial weights will be matrix. Let's discuss the proper api using `params` framework later.